app/components/post_list/load_more_button.js

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2017 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React, {PureComponent, PropTypes} from 'react';
```

Comment:
@enahum We discussed this in a call, but I'll note my reasoning here too:
This component serves two functions: a loading indicator and a button to load more.

1) Usually when scrolling up the posts list, the user will not reach the top before the next page is loaded, but in case they do (maybe the server is slow), we display a loading indicator.

2) We're using complex, somewhat brittle logic for determining if we've loaded the channel's initial posts, if we've loaded more posts, and if we've reached the very first post. It is plausible that the user could encounter an edge case or corrupted state where our logic doesn't hold up. In that unlikely event, we're displaying a button which allows them to manually load more posts.